src/tim/transform/ops/conv2d_layout_inference.h

```diff
@@ -40,63 +40,89 @@ class Conv2dLayoutInfer : public OpLayoutInfer {
       : OpLayoutInfer(op, context) {}
   void OnInputs(
       std::vector<std::shared_ptr<vx::Tensor>>& next_tensors) override {
+    auto src_conv2d = std::static_pointer_cast<vx::ops::Conv2d>(op_);
     vx::DataLayout layout = op_->impl()->layout_;
-    auto required_pv = MakeShared(4);
-    if (layout == vx::DataLayout::CWHN) {
-      required_pv = std::make_shared<PermuteVector<4>>(kCWHN2WHCN);
+    auto kernel_layout = src_conv2d->KernelDataLayout();
+    std::shared_ptr<IPermuteVector> required_pv, weight_required_pv, trans_pv;
+    switch (layout)
+    { // kernel layout must be IWHO in tflite & nnapi
+      case vx::DataLayout::CWHN:
+        required_pv = std::make_shared<PermuteVector<4>>(kCWHN2WHCN);
+        break;
+      case vx::DataLayout::WHCN:
+        required_pv = MakeShared(4);
+        break;
+      default:
+        VSILOGE("The layout of input is not support.");
+        required_pv = MakeShared(4);
+        break;
+    }
+    switch (kernel_layout) {
+      case vx::DataLayout::OcIcWH:  // Support TVM Kernel Layout
+        weight_required_pv = std::make_shared<PermuteVector<4>>(kOcIcWH2WHIcOc);
+        break;
+      case vx::DataLayout::IcOcWH:
+        weight_required_pv = std::make_shared<PermuteVector<4>>(kIcOcWH2WHIcOc);
+        break;
+      case vx::DataLayout::IcWHOc:  // Support nnapi & tflite Kernel Layout
+        weight_required_pv = std::make_shared<PermuteVector<4>>(kIcWHOc2WHIcOc);
+        break;
+      default: // Default set to IWHO for compatibility with previous APIs
+        weight_required_pv = std::make_shared<PermuteVector<4>>(kIcWHOc2WHIcOc);
+        break;
     }
-    auto input_tensors = op_->impl()->InputsTensor();
 
+    auto input_tensors = op_->impl()->InputsTensor();
+    trans_pv = required_pv;
     for (const auto& in : input_tensors) {
```

Comment:
why not handle it one by one without for-loop? If so, the code may more clean than now.

Reply:
done